Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItemDefault.swift

```diff
@@ -0,0 +1,208 @@
+ */
+
+import Foundation
+
+internal class BPKHorizontalNavigationItemDefault<Size: BPKHorizontalNavigationSize>: UIButton, BPKHorizontalNavigationItem {
+    struct IconDefinition {
+        let image: UIImage
+        let size: CGSize
+    }
+
+    override var isSelected: Bool {
+        didSet {
+            updateStyle()
+        }
+    }
+
+    dynamic var selectedColor: UIColor? {
+        didSet {
+            updateStyle()
+        }
+    }
+
+    var appearance: BPKHorizontalNavigationAppearance = .normal {
+        didSet {
+            updateStyle()
+        }
+    }
+
+    var icon: IconDefinition?
+
+    var hasIcon: Bool { icon != nil }
+    
+
+    /**
+     * The name to display within the item.
+     */
+    let name: String
+
+    private(set) var showNotificationDot: Bool = false
+
+    private lazy var notificationDot: UIView = {
+        let dot = UIView()
+        dot.translatesAutoresizingMaskIntoConstraints = false
+        dot.backgroundColor = .bpk_systemRed
+        dot.layer.cornerRadius = BPKSpacingMd / 2.0
+
+        return dot
+    }()
+
+    private var contentColor: UIColor {
+        if isSelected {
+            return selectedColor ?? .bpk_primary
+        } else {
+            switch (appearance) {
+            case .normal: return .bpk_textPrimary
+            case .alternate: return .bpk_skyGrayTint07
+            }
+        }
+    }
+
+    init(name: String) {
+        self.name = name
+
+        super.init(frame: .zero)
+
+        setUp()
+    }
+
+    init(name: String, showNotificationDot: Bool) {
+        self.name = name
+        self.showNotificationDot = showNotificationDot
+
+        super.init(frame: .zero)
+
+        if (showNotificationDot) {
+            addSubview(notificationDot)
+        }
+
+        setUp()
+    }
+
+    init(name: String, iconDefinition: IconDefinition) {
+        self.name = name
+        self.icon = iconDefinition
+
+        super.init(frame: .zero)
+
+        setUp()
+    }
+
+    init(name: String, iconDefinition: IconDefinition, showNotificationDot: Bool) {
+        self.name = name
+        self.icon = iconDefinition
+        self.showNotificationDot = showNotificationDot
+
+        super.init(frame: .zero)
+
+        if (showNotificationDot) {
+            addSubview(notificationDot)
+        }
+
+        setUp()
+    }
+
+
+    override var intrinsicContentSize: CGSize {
+        let buttonIntrinsicContentSize = super.intrinsicContentSize
+
+        if let icon = icon {
+            let iconHeight = icon.size.height
+            let width = buttonIntrinsicContentSize.width + 3 * Size.horizontalSpacing
+            let height = iconHeight + Size.verticalSpacing / 2.0
+
+            return CGSize(width: width, height: height)
+        } else {
+            let width = buttonIntrinsicContentSize.width + 2 * Size.horizontalSpacing
+            let height = buttonIntrinsicContentSize.height
+
+            return CGSize(width: width, height: height)
```

Comment:
I think the else is not needed and can be written like this
```swift 
if let icon = icon {
   let iconHeight = icon.size.height
   let width = buttonIntrinsicContentSize.width + 3 * Size.horizontalSpacing
   let height = iconHeight + Size.verticalSpacing / 2.0
 
   return CGSize(width: width, height: height)
}
let width = buttonIntrinsicContentSize.width + 2 * Size.horizontalSpacing
let height = buttonIntrinsicContentSize.height
 
return CGSize(width: width, height: height)
```
or even using a `guard let icon = icon else { return without icon logic }`